### ISSUE TITLE
I cannot find the requirements.txt file

### ISSUE THREAD

**Issue by alelasantillan** [areed1192/td-ameritrade-api#2]
I think that the requirements file is missing. Am I wrong?
When I try to run a test like this:

```terminal
(myenv3.7.3) (base) Alejandro-2:tests alejandrosantillaniturres$ python test_client.py 
Traceback (most recent call last):
  File "test_client.py", line 4, in <module>
    from td.credentials import TdCredentials
ModuleNotFoundError: No module named 'td'
```

I did:

```terminal
pip install td
```

and a module installed, but it is not the module the test_client.py is requesting. Which is the correct module to install? 

I did:

```terminal
pip install tdameritrade
```

but again, is not the right module.  Please help!

**Comment by areed1192** [comment]
I just added the `requirements.txt` file so you should be able to install the dependencies that are required in the library. I would recommend that if you currently have the old TD library install on your system, that you either uninstall it or create virtual environment that you can control which packages are there.

**Comment by alelasantillan** [comment]
Hi areed1192, I did:
pip install -r requirements.txt
pip install . 

everything went fine and then:
cd test/
python test_client.py

And I got:

Traceback (most recent call last):
  File "test_client.py", line 5, in <module>
    from td.credentials import TdCredentials
ModuleNotFoundError: No module named 'td'

Which is the td missing module I have to install?


Then I tried:

I also got an error when trying to run the sample code on your README.md file:

Traceback (most recent call last):
  File "sample_code.py", line 14, in <module>
    client_id = config.get('main', 'client_id')
  File "~/.pyenv/versions/3.8.8/lib/python3.8/configparser.py", line 781, in get
    d = self._unify_values(section, vars)
  File "~/.pyenv/versions/3.8.8/lib/python3.8/configparser.py", line 1149, in _unify_values
    raise NoSectionError(section) from None
configparser.NoSectionError: No section: 'main'

Can you tell me what to do?

**Comment by areed1192** [comment]
The issue is you don't have a config file that stores all your credentials like your client ID and redirect URI. Here's an example of how to create a config file.

```python
from configparser import ConfigParser

# Initialize the Parser.
config = ConfigParser()

# Add the Section.
config.add_section('main')

# Set the Values.
config.set('main', 'client_id', 'YOUR_CLIENT_ID')
config.set('main', 'redirect_uri', 'YOUR_REDIRECT_URI')
config.set('main', 'account_number', 'YOUR_ACCOUNT_NUMBER')

# Write the file.
with open(file='config/config.ini', mode='w+') as f:
    config.write(f)
```

This will create a `config.ini` file in a folder called `config` if it exists. You need to make sure that the config folder exists before running this code.

**Comment by alelasantillan** [comment]
Thank you very much Alex!

I run this script and worked fine.
Now, I need to provide the values:
> YOUR_CLIENT_ID, which is the CONSUMER_KEY that td ametritrade asigns after registering a new app
> YOUR_REDIRECT_URI, which is localhost
> YOUR_ACCOUNT_NUMBER

I’ve created a developer acct in developer.tdameritrade.com and
I’ve registered a new app, so I’ve got a CONSUMER_KEY for that app. I think that the client_id is the CONSUMER_KET,  but I don’t know/understand how to get my ACCOUNT_NUMBER. I’ve just have my username/password
There seems to be no such information anywhere in the developer area.
Can you guide me a little more. I am a bit confused.
Thank you,
Alejandro

**Comment by areed1192** [comment]
The `account_number` is just your TD Ameritrade account number that you see when you login to your account. Naturally, if you want to access account information, you'll need your TD Ameritrade account number. So any of the endpoints where you need to provide the account number, you need to pass it through.

**Comment by alelasantillan** [comment]
All right @areed1192 . So it seems that one will have to log in manually since there is no relation between the ACCOUNT_NUMBER and the CONSUMER_KEY. Do you know if there is there any way to pass credentials automatically.
Otherwise one cannot have a fully automatic tradingbot working against TD ameritrade.

**Comment by cmpunches** [comment]
Their api will let you list the accounts you have along with their IDs.

As to the manual login, it will be required once every 90 days.  That's an oauth2 design defect.  You can blame corporate architects padding their resumes with security theatre at the expense of design coherency and user experience for that.  Can't be fixed unless they drop oAuth on the server side.

**Comment by areed1192** [comment]
@alelasantillan Just to reiterate the same topic, the `oAuth` process must be done every 90 days regardless of if we want to or not. Now there are ways to automate it using selenium, but I will not be including that functionality into the library since selenium requires different versions of the driver depending on the browser.

Also, I might be speaking to early, but I think something you could try is to refresh the `refresh_token` before the 90-day expiration date and see if that allows you to bypass the full `oAuth` process.

**Comment by alelasantillan** [comment]
Thank you @areed1192 and @cmpunches!

Now I've created the config dir, and run the create_config code above. It created the config/config.ini files which contains:

  1 [main]
  2 client_id = YOUR_CLIENT_ID
  3 redirect_uri = YOUR_REDIRECT_URI
  4 account_number = YOUR_ACCOUNT_NUMBER

But when I run: 
**python sample_code.py**
(sample_code.py is the script you posted in the README.md into the Usage section)

I get:
    **with open(file=file_path, mode='r') as token_file:
FileNotFoundError: [Errno 2] No such file or directory: 'config/td_credentials.json'**

So I renamed the config.ini file to td_credentials.json and, logically, the script chimes about the non-jason format.
    **raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 2 (char 1)**

So I changed the format to read:
  1 {
  2    "main":[
  3       {
  4           "client_id" : "YOUR_CLIENT_ID",
  5           "redirect_uri" : "YOUR_REDIRECT_URI",
  6           "account_number" : "YOUR_ACCOUNT_NUMBER"
  7       }   
  8    ]  
  9 }  
  
  And replaced the client_id by the TD's consumer key, the http://localhost and the tdameritrade.com acct number and I run:
    **python sample_code.py** 
  And I get (in terminal):
**Refresh Token Expired, initiating oAuth workflow.
Please Paste the Authorization Code Here:**       

And also a blank window in my web browser opened with this address:
https://auth.tdameritrade.com/%27http://localhost'?error=invalid_client&error_description=invalid_client

Any idea?

**Comment by areed1192** [comment]
Can you run the code in this [file](https://github.com/areed1192/td-ameritrade-api/blob/master/samples/use_ouath.py). It should run the oAuth process for you, let me know if you run into any issues with this file. Also, remember to check the file paths and that they all match.

**Comment by alelasantillan** [comment]
Hi @areed1192 !!!
thank you very much. I was able to get the refresh and auth tokens. They are the following (I've replaced the true codes a bit to avoid security risks). 

I am now wondering if all this json output should be put in the credential_file='config/td_credentials.json' as is, or if only access_token and refresh_token needs to be included. 

Also I wonder if case of access_token and resfresh_token should be changed to uppercase since it seems that the file I have contains:

{"access_token": "vXN8KDsldkfjalsdkfjFWU+s1P8qTReu2/WaZ7X/OmamaQRcIhehjbKPFxu+ntWjZPcgJHmHH6jmyi48MyfjOp5DyjsdJQRKh7XQh673c+P8p7ocXMCgUdsfj2xJHmD1SiTLXc4LmCXzo/tQFZU/kqwl6vRzWqX2Zu3F1qshJriShAE2VQ2ZloscyeohvOQRpKWwFmEllxqBPDODAeA1dggwHJY3UakPhzPnBIUrtHDiPKHc3i3W+Hk2SUqRH7VP6duvYZ3t/0acM42A+g0ovCXppwyinz3X4VIxh9QCmf2c5XHWuIhjDKjmh1guOZKHs+fZeQ1A1CAWfdhswbOOkSxkQYIi4828wK100MQuG4LYrgoVi/JHHvl2NJ7KxzgsKey/aKAKNKsyCa1J/QuXJcQ0kx0Zrpyx0MAcOXZQmYN4Y8H6jhj/5mbLaLdqzBXa637kFxI1nnPMCjdF0Ya9hx52FBZV9RjQnk1bH5N3j3bpv5RvACVet09Nm2THOH8VM9/9hflLqkFjqlnBFndiTcyRDBtdcGmuIHmQW/+jlfeeiIEISNOSIOFOIS5E", "refresh_token": "uKBoCpFhyPaaS1GuQ2/i3cov1zpcIs+PTA5g6xJhNzlY/eMmAH6M5RA0on91fWSoFeCvXI+zk4GHV1ihAijsTAPJVjlHifmdaQtAglBSNkag/8+lYvAGSsq+06v/JMZQ7vJ4xU9HnMBfiC4YdDtIHT6e8sN7XD3sGP7IXsSi59rI/T7BAKW4aNkz8nkWkrPSESVLodXr9IvallulFiTQFBHHiW8gC35MufaCHUXZBRFBi+RWUO/OyW6Vh1OdPXmK5wmq5NZPmDpMMU6+zF+t323BWclr3+CmV/WvKUnzDzGwuB+IDI8yUAdhPtfuePO4IxbV/ACt4QJdUCyDmhRYUQX7h8zJo2y5z1rPdminUKGdmdw=212FD3x19z9sWBHDJACbC00B75E", "scope": "PlaceTrades AccountAccess MoveMoney", "expires_in": 1800, "refresh_token_expires_in": 7776000, "token_type": "Bearer"}
